### PR TITLE
Accept YAML as input

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-1898

### CODE REVIEW

File: cmd/validate/image_test.go [hunk 1]
```diff
@@ -122,6 +123,39 @@ func Test_determineInputSpec(t *testing.T) {
...
+                ---
+                application: app1
+                components:
+                - name: nodejs
+                  containerImage: quay.io/hacbs-contract-demo/single-nodejs-app:877418e
+                - name: petclinic
+                  containerImage: quay.io/hacbs-contract-demo/spring-petclinic:dc80a7f
+                - name: single-container-app
+                  containerImage: quay.io/hacbs-contract-demo/single-container-app:62c06bf
+                `),
```
Comment (simonbaird) [inline cmd/validate/image_test.go:139]: Indent looks off, but I'm guessing it's spaces instead of tabs. Consider putting real tabs in maybe?

Comment (lcarva) [reply cmd/validate/image_test.go:139]: Tabs are not allowed in YAML.

Comment (simonbaird) [reply cmd/validate/image_test.go:139]: Oh, I did not know that..

File: cmd/validate/image_test.go [hunk 2]
```diff
@@ -79,12 +80,12 @@ func Test_determineInputSpec(t *testing.T) {
 		{
 			name: "faulty ApplicationSnapshot string",
 			arguments: data{
-				input: "/",
+				input: "{",
 			},
-			err: "invalid character '/' looking for beginning of value",
+			err: "unable to parse ApplicatonSnapshot from input: error converting YAML to JSON: yaml: line 1: did not find expected node content",
```
Comment (simonbaird) [inline cmd/validate/image_test.go:85]: Typo maybe?

```suggestion
			err: "unable to parse ApplicationSnapshot from input: error converting YAML to JSON: yaml: line 1: did not find expected node content",
```

**Review by simonbaird — APPROVED** [review]

File: cmd/validate/image_test.go (hunk 1 above)
Comment (lcarva) [reply cmd/validate/image_test.go]: Well.. it turns out that `hd.Doc` does remove the leading tabs. Updated.

File: cmd/validate/image_test.go (hunk 2 above)
Comment (lcarva) [reply cmd/validate/image_test.go]: Yes. Thank you. Updated!

File: cmd/validate/image_test.go (hunk 1 above)
Comment (simonbaird) [reply cmd/validate/image_test.go]: Ah yes. The unusual case where some tabs followed by some spaces is correct. :smile: .

**Pull request merged**